airtable/plugin.py

```diff
@@ -83,21 +113,22 @@ def _resolve_error_message(
 
 @dataclass
 class AirtableRecord:
-    id: str
+    airtable_record_id: AirtableRecordID
     created_time: datetime
     fields: dict[str, Any]
 
 
-def airtable_record_list(base_id: str, table_id_or_name: str) -> list[AirtableRecord]:
+def airtable_record_list(airtable_base_id: AirtableBaseID, airtable_table_id: AirtableTableID) -> list[AirtableRecord]:
     """
     Return results of an Airtable `list records` API call.
-    base_id must be an ID and not a name.
     """
+    base_id = airtable_base_id.id
+    table_id = airtable_table_id.id
```

Comment:
If we end up renaming the arguments to `base_id` and `table_id`, I suggest using them directly instead of aliasing.